### PR TITLE
[AArch64] Fix a presumed typo in isFPImmLegal limit. NFC

### PR DESCRIPTION
**Pull request by citymarina**
The worst possible case for a double literal goes like:

```
  mov ...
  movk ..., lsl #16
  movk ..., lsl #32
  movk ..., lsl #48
  fmov ...
```

The limit of 5 in the code gives the impression that  `Insn` includes all instructions including the `fmov`, but that's not true. It only counts the integer moves. This led me astray on some other work in this area.

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 1]
```diff
@@ -11463,7 +11463,8 @@ bool AArch64TargetLowering::isFPImmLegal(const APFloat &Imm, EVT VT,
     // movw+movk is fused). So we limit up to 2 instrdduction at most.
     SmallVector<AArch64_IMM::ImmInsnModel, 4> Insn;
     AArch64_IMM::expandMOVImm(ImmInt.getZExtValue(), VT.getSizeInBits(), Insn);
-    unsigned Limit = (OptForSize ? 1 : (Subtarget->hasFuseLiterals() ? 5 : 2));
+    assert(Insn.size() <= 4);
```
Comment (fhahn) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:11466]: Would be good to add a message to the assert(`&& "...")`

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 2]
```diff
@@ -11463,7 +11463,9 @@ bool AArch64TargetLowering::isFPImmLegal(const APFloat &Imm, EVT VT,
     // movw+movk is fused). So we limit up to 2 instrdduction at most.
```
Comment (davemgreen) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:11463]: Do you mind changing instrdduction to instructions whilst you are here?

**Review by davemgreen — APPROVED** [review]
LGTM, thanks

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 2 above)
Comment (davemgreen) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:11463]: Oh - you remove it in a follow up. Don't worry about it here then.

**Review by fhahn — APPROVED** [review]
LGTM, thanks

**Pull request merged**